File: sdks/dotnet/src/Resources/ProductsResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class ProductsResource
    {
        private readonly PuxbayClient _client;

        public ProductsResource(PuxbayClient client)
        {
            _client = client;
        }

        public async Task<PaginatedResponse<Product>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await _client.GetAsync<PaginatedResponse<Product>>($"products/?page={page}&page_size={pageSize}");
        }

        public async Task<Product> GetAsync(string productId)
        {
            return await _client.GetAsync<Product>($"products/{productId}/");
        }

        public async Task<Product> CreateAsync(Product product)
        {
            return await _client.PostAsync<Product>("products/", product);
        }

        public async Task<Product> UpdateAsync(string productId, Product product)
        {
            return await _client.PatchAsync<Product>($"products/{productId}/", product);
        }

        public async Task DeleteAsync(string productId)
        {
            await _client.DeleteAsync($"products/{productId}/");
        }
    }
}

File: sdks/php/src/Resources/ProductsResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Product;
use Puxbay\Models\PaginatedResponse;

class ProductsResource extends BaseResource
{
    public function list(int $page = 1, int $pageSize = 20): PaginatedResponse
    {
        $data = $this->get("products/?page={$page}&page_size={$pageSize}");
        return PaginatedResponse::fromArray($data, fn($item) => Product::fromArray($item));
    }

    public function getProduct(string $productId): Product
    {
        $data = $this->get("products/{$productId}/");
        return Product::fromArray($data);
    }

    public function create(Product $product): Product
    {
        $data = $this->post('products/', $product->toArray());
        return Product::fromArray($data);
    }

    public function update(string $productId, Product $product): Product
    {
        $data = $this->patch("products/{$productId}/", $product->toArray());
        return Product::fromArray($data);
    }

    public function deleteProduct(string $productId): void
    {
        $this->delete("products/{$productId}/");
    }

    public function adjustStock(string $productId, int $quantity, string $reason, ?string $notes = null): Product
    {
        $payload = [
            'quantity' => $quantity,
            'reason' => $reason,
        ];
        if ($notes) {
            $payload['notes'] = $notes;
        }

        $data = $this->post("products/{$productId}/stock-adjustment/", $payload);
        return Product::fromArray($data);
    }
}
